test/data/s3/test_s3.py

```diff
@@ -859,15 +860,27 @@ def test_put_one(s3root, objs, expected):
     argnames=["s3root", "blobs", "expected"], **s3_data.pytest_put_blobs_case()
 )
 def test_put_files(tempdir, inject_failure_rate, s3root, blobs, expected):
+    def _write_file_py35(path, data):
+        with open(path, "ab") as f:
+            for d in data.split(b"\n"):
+                f.write(d)
+
+    current_py_version = int(platform.python_version_tuple()[0]) * 10 + int(
+        platform.python_version_tuple()[1]
+    )
+
     def _files(blobs):
         for blob in blobs:
             key = getattr(blob, "key", blob[0])
             data = getattr(blob, "value", blob[1])
             content_type = getattr(blob, "content_type", None)
             metadata = getattr(blob, "metadata", None)
             path = os.path.join(tempdir, key)
-            with open(path, "wb") as f:
-                f.write(data)
+            if current_py_version < 36:
+                _write_file_py35(path, data)
+            else:
+                with open(path, "wb") as f:
+                    f.write(data)
```

Comment:
Had to add all this to tests so that large files can be written safely with py3.5. Otherwise it resulted in the following errors in the tests : 

```
=================================== FAILURES ===================================
__________________________ test_put_files[5gb_file-0] __________________________

tempdir = './metaflow.test.tmpipu61zea', inject_failure_rate = 0
s3root = '***put_tests'
blobs = [('1135d924-d320-4efb-b0c2-1918477df279', b'\xffLi9C[\xb6\x94\xa8\xe8\xa7\xa1V\x1b\xa8\xd7\xc0\x00\x02\xafL\x90\xf0\x0...6\xfcKI\xaa\xfd\x1c8\xd15\xbdax\xa0\x11u(;\xbd!K\n\x19\xb5\x01u,\x8b\xac\xf7\xa6WE\xe4\x93a\xb8\xfcr\xccP6\x15\x1eH~')]
expected = {'***put_tests/1135d924-d320-4efb-b0c2-1918477df279': {None: ExpectedResult(size=5368709120, checksum='c69d7ec01fbb904bc200c58dff813ad68253e003', content_type=None, metadata=None, range=None)}}

    @pytest.mark.parametrize("inject_failure_rate", [0, 10, 50, 90])
    @pytest.mark.parametrize(
        argnames=["s3root", "blobs", "expected"], **s3_data.pytest_put_blobs_case()
    )
    def test_put_files(tempdir, inject_failure_rate, s3root, blobs, expected):
        def _files(blobs):
            for blob in blobs:
                key = getattr(blob, "key", blob[0])
                data = getattr(blob, "value", blob[1])
                content_type = getattr(blob, "content_type", None)
                metadata = getattr(blob, "metadata", None)
                path = os.path.join(tempdir, key)
                with open(path, "wb") as f:
                    f.write(data)
                yield S3PutObject(
                    key=key, value=path, content_type=content_type, metadata=metadata
                )
    
        with S3(s3root=s3root, inject_failure_rate=inject_failure_rate) as s3:
>           s3urls = s3.put_files(_files(blobs))

s3/test_s3.py:876: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../metaflow/plugins/datatools/s3/s3.py:1247: in put_files
    return self._put_many_files(_check(), overwrite)
../../metaflow/plugins/datatools/s3/s3.py:1327: in _put_many_files
    url_info = list(url_info)
../../metaflow/plugins/datatools/s3/s3.py:1227: in _check
    for key_path in key_paths:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

blobs = [('1135d924-d320-4efb-b0c2-1918477df279', b'\xffLi9C[\xb6\x94\xa8\xe8\xa7\xa1V\x1b\xa8\xd7\xc0\x00\x02\xafL\x90\xf0\x0...6\xfcKI\xaa\xfd\x1c8\xd15\xbdax\xa0\x11u(;\xbd!K\n\x19\xb5\x01u,\x8b\xac\xf7\xa6WE\xe4\x93a\xb8\xfcr\xccP6\x15\x1eH~')]

    def _files(blobs):
        for blob in blobs:
            key = getattr(blob, "key", blob[0])
            data = getattr(blob, "value", blob[1])
            content_type = getattr(blob, "content_type", None)
            metadata = getattr(blob, "metadata", None)
            path = os.path.join(tempdir, key)
            with open(path, "wb") as f:
>               f.write(data)
E               OSError: [Errno 22] Invalid argument

s3/test_s3.py:870: OSError
=============================== warnings summary ===============================
```